packages/scm/src/browser/scm-widget.tsx

```diff
@@ -485,6 +488,30 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
         return [this.props.resource];  // TODO support multiselection
     }
 
+    /**
+     * Handle the single clicking of nodes present in the widget.
+     */
+    protected handleClick = () => {
+        // Determine the behavior based on the preference value.
+        const isSingle = this.props.corePreferences && this.props.corePreferences['workbench.list.openMode'] === 'singleClick';
+        if (isSingle) {
+            this.open();
+        } else {
+            this.selectChange();
```

Comment:
I think it should happen always.

Reply:
I've updated it to always perform `selectChange` as mentioned.